src/models/server.ts

```diff
@@ -0,0 +1,14 @@
+import { BaseModel } from "./base";
+import { ServerSecurityRequirementInterface } from "./server-security-requirement";
+import { ServerVariableInterface } from "./server-variables";
+import {BindingsMixinInterface, DescriptionMixinInterface, SpecificationExtensionsMixinInterface} from './mixins';
+
+export interface ServerInterface extends BaseModel, BindingsMixinInterface, DescriptionMixinInterface, SpecificationExtensionsMixinInterface {
+    url(): string;
+    protocol(): string;
+    protocolVersion(): string
+    variables(): Record<string, ServerVariableInterface>;
+    variable(name: string): ServerVariableInterface;
+    hasVariables(): boolean;
+    security(): [ServerSecurityRequirementInterface] | undefined;
```

Comment:
`url` and `protocol` are required, rest of fields are optional, so they should return `... | undefined`.

Also I opt to have methods like:

```ts
hasVariables(): boolean;
hasVariables(name: string): boolean;

variables(): Record<string, ServerVariableInterface> | undefined;
variables(name: string): ServerVariableInterface | undefined;
```

cc @smoya 

also type like `[ServerSecurityRequirementInterface]` is a tuple, not array of the items.

Reply:
Please also read https://github.com/asyncapi/parser-js/pull/497#discussion_r830002712.

Intermediate models are the solution.